Upgraded to Clang 11 and now see the -Wdangling warning mentioned in the PR description.
```
wallet/rpcwallet.cpp:42:22: warning: temporary whose address is used as value of local variable
'a_bad_vec' will be destroyed at the end of the full-expression [-Wdangling]
Span<const int> a_bad_vec(std::vector<int>{1,2,3});
                     ^~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

--- a/src/attributes.h
+++ b/src/attributes.h
@@ -19,4 +19,14 @@
 #  endif
 #endif
 
+#if defined(__clang__)
+#  if __has_attribute(lifetimebound)
+#    define LIFETIMEBOUND [[clang::lifetimebound]]
+#  else
+#    define LIFETIMEBOUND
+#  endif
+#else
+#  define LIFETIMEBOUND
+#endif
+
 #endif // BITCOIN_ATTRIBUTES_H

